src/illuminatio/rule.py

```diff
@@ -18,17 +21,29 @@
 
 
 def dict_mapper(obj):
+    """
+    converts an object to a dictionary if possible
+    """
     return obj.to_dict() if hasattr(obj, "to_dict") else obj
 
 
 class Connection(Conn):
+    """
+    class for interaction with Connection tuples
```

Comment:
Start with a capital letter to match the rest of the docstrings (see: https://www.python.org/dev/peps/pep-0257/)